tests/client/test_client.py

```diff
@@ -531,6 +533,304 @@ def test_epi_create(self):
+        adjustment_factor_payload["name"] = "modified name"
+        adjustment_factor_result = epi_client.update_adjustment_factor(
+            adjustment_factor_id, adjustment_factor_payload
+        )
+        common_result_check(adjustment_factor_result, adjustment_factor_payload, "name")
+
+        # read
+        adjustment_factor_result = epi_client.get_adjustment_factor(adjustment_factor_id)
+        common_result_check(adjustment_factor_result, adjustment_factor_payload, "name")
+
+        ###
+        ### data extraction
+        ###
+        data_extraction_payload = {
+            "design": design_id,
+            "outcome_id": outcome_id,
+            "exposure_level_id": exposure_level_id,
+            "sub_population": "DE subpop",
+            "outcome_measurement_timing": "DE meas timing",
+            "effect_estimate_type": "Absolute Risk %",
+            "effect_estimate": 1.1,
+            "ci_lcl": 1.2,
+            "ci_ucl": 1.3,
+            "ci_type": "Rng",
+            "units": "DE units",
+            "variance_type": "SD",
+            "variance": 1.4,
+            "n": 99,
+            "p_value": "DE pval",
+            "significant": "No",
+            "group": "DE group",
+            "exposure_rank": 1,
+            "exposure_transform": "DE expo transform",
+            "outcome_transform": "DE outcome transform",
+            "factors": adjustment_factor_id,
+            "confidence": "DE confidence",
+            "data_location": "DE loc",
+            "effect_description": "DE effect desc",
+            "statistical_method": "DE stat method",
+            "comments": "DE comments",
+        }
+
+        # create
+        data_extraction_result = epi_client.create_data_extraction(data_extraction_payload)
+        common_result_check(data_extraction_result, data_extraction_payload, "comments")
+
+        data_extraction_id = data_extraction_result["id"]
+
+        # update
+        data_extraction_payload["comments"] = "modified comments"
+        data_extraction_result = epi_client.update_data_extraction(
+            data_extraction_id, data_extraction_payload
+        )
+        common_result_check(data_extraction_result, data_extraction_payload, "comments")
+
+        # read
+        data_extraction_result = epi_client.get_data_extraction(data_extraction_id)
+        common_result_check(data_extraction_result, data_extraction_payload, "comments")
+
+        ###
+        ### all
+        ###
+        # delete
+        # tuple of tuples, each sub-tuple is:
+        # [0] a primary key
+        # [1] the client method to use to delete it
+        # [2] the model class
+
+        deletions = (
+            (data_extraction_id, epi_client.delete_data_extraction, epiv2models.DataExtraction),
+            (exposure_level_id, epi_client.delete_exposure_level, epiv2models.ExposureLevel),
+            (outcome_id, epi_client.delete_outcome, epiv2models.Outcome),
+            (
+                adjustment_factor_id,
```

Comment:
shorten names so ruff formats to a single line

Reply:
done